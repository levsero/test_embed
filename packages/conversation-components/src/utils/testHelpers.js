import { render as rtlRender } from '@testing-library/react'
import ThemeProvider from 'src/ThemeProvider'

export const render = (ui, { render, themeProps = {} } = {}) => {
  const renderFn = render || rtlRender
  return {
    ...renderFn(<ThemeProvider theme={themeProps}>{ui}</ThemeProvider>)
  }
}
