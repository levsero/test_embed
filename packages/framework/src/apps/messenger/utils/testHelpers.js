import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { Provider } from 'react-redux'
import ThemeProvider from 'src/apps/messenger/features/themeProvider'
import createStore from 'src/apps/messenger/store'
import hostPageWindow from 'src/framework/utils/hostPageWindow'

export const render = (ui, { render, store, themeProps = {} } = {}) => {
  const reduxStore = store || createStore()

  const renderFn = render || rtlRender
  return {
    store: reduxStore,
    ...renderFn(
      <Provider store={reduxStore}>
        <ThemeProvider theme={themeProps}>{ui}</ThemeProvider>
      </Provider>
    )
  }
}

export const mockMatchMedia = () => {
  // Keep track of all media queries made and their callbacks
  // {
  //   <media query>: [ <callback1>, <callback2> ]
  // }
  const mediaQueries = {}

  const mockMatchMedia = jest.fn().mockImplementation(query => {
    if (!mediaQueries[query]) {
      mediaQueries[query] = []
    }

    return {
      matches: true,
      addEventListener: (_, callback) => {
        mediaQueries[query].push(callback)
      }
    }
  })

  const triggerChangeForBreakpoint = (breakpoint, event) => {
    mediaQueries[breakpoint]?.forEach(callback => callback(event))
  }

  hostPageWindow.matchMedia = mockMatchMedia

  return {
    triggerChangeForBreakpoint
  }
}

/*
  For testing reducers (duh).
  Pass it your reducer and an array of parameter objects.
  - If parameter object is a flat action payload, the test will use
  snapshot testing. The action can also be explicitly specified using
  an `action` key.
  - If parameter object contains key `initialState`, the value
  of that key will be used as initial state for the reducer.
  - If parameter object contains key `expected`, the value
  of that key will be used to assert against the result of the reducer.
*/
export const testReducer = (reducer, actions) => {
  actions.forEach(params => {
    const { expected, initialState, extraDesc } = params
    const action = params.action || params
    const basicTestDesc = `${reducer.name}, action: ${action.type}`
    const testDesc = extraDesc ? `${basicTestDesc} ${extraDesc}` : basicTestDesc

    test(testDesc, () => {
      const reduced = reducer(initialState, action)

      if ('expected' in params) {
        expect(reduced).toEqual(expected)
      } else {
        expect(reduced).toMatchSnapshot()
      }
    })
  })
}
