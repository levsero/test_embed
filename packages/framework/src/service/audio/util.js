import { includes } from 'lodash'

const canPlayTypeResultWhitelist = ['maybe', 'probably']

export function getSupportedAudioType(audio) {
  if (!audio.canPlayType) {
    return
  }

  const canPlayType = type => includes(canPlayTypeResultWhitelist, audio.canPlayType(type))

  if (canPlayType('audio/mpeg')) {
    return 'mp3'
  } else if (canPlayType('audio/ogg; codecs="vorbis"')) {
    return 'ogg'
  } else if (canPlayType('audio/wav')) {
    return 'wav'
  }
}
