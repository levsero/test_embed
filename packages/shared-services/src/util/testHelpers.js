export const clearDOM = () => {
  document.getElementsByTagName('html')[0].innerHTML = ''
}
