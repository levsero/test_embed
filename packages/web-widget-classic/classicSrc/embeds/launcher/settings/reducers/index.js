import chatLabel from 'classicSrc/embeds/launcher/settings/reducers/chatLabel'
import label from 'classicSrc/embeds/launcher/settings/reducers/label'
import talkLabel from 'classicSrc/embeds/launcher/settings/reducers/talkLabel'
import { combineReducers } from 'redux'

export default combineReducers({
  label,
  chatLabel,
  talkLabel,
})
