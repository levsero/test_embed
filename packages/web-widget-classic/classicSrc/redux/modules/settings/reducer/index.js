import { combineReducers } from 'redux'
import analytics from './analytics'
import answerBot from './answerBot'
import chat from './chat'
import color from './color'
import contactForm from './contactForm'
import contactOptions from './contactOptions'
import cookies from './cookies'
import helpCenter from './help-center'
import launcher from './launcher'
import navigation from './navigation'
import styling from './styling'
import talk from './talk'

export default combineReducers({
  analytics,
  answerBot,
  chat,
  color,
  cookies,
  contactForm,
  contactOptions,
  helpCenter,
  launcher,
  navigation,
  styling,
  talk,
})
