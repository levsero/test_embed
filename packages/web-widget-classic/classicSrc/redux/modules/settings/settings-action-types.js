export const UPDATE_SETTINGS = 'widget/settings/UPDATE_SETTINGS'
export const ALL_FORMS_REQUESTED = 'widget/settings/ALL_FORMS_REQUESTED'
