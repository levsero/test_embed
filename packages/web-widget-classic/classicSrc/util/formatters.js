import { i18n } from 'classicSrc/app/webWidget/services/i18n'
import _ from 'lodash'

const defaultOptions = {
    showToday: false,
    month: 'long',
    day: 'numeric',
    year: 'numeric',
    hour: 'numeric',
    minute: 'numeric',
  },
  minuteOptions = {
    hour: 'numeric',
    minute: 'numeric',
  }

export function dateTime(timestamp, options = {}) {
  const mergedOpts = _.merge({}, defaultOptions, options)
  const messageFormattedDate = i18n.dateTimeFormat(timestamp, mergedOpts)

  if (mergedOpts.showToday && isToday(timestamp)) {
    const localMinutes = i18n.dateTimeFormat(timestamp, minuteOptions)

    return i18n.t('embeddable_framework.common.today', {
      time: localMinutes,
    })
  }

  return messageFormattedDate
}

const isToday = (inStamp) => {
  const inDate = new Date(inStamp),
    today = new Date()

  return (
    inDate.getDate() === today.getDate() &&
    inDate.getMonth() === today.getMonth() &&
    inDate.getFullYear() === today.getFullYear()
  )
}
