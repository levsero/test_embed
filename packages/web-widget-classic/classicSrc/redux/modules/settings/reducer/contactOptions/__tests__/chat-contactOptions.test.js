import { UPDATE_SETTINGS } from 'classicSrc/redux/modules/settings/settings-action-types'
import { testReducer } from 'classicSrc/util/testHelpers'
import chatContactOptions from '../chat-contactOptions'

const badPayload = {
  foo: 'bar',
}
const goodPayload = {
  webWidget: {
    contactOptions: {
      chatLabelOnline: { '*': 'online label' },
      chatLabelOffline: { '*': 'offline label' },
    },
  },
}

testReducer(chatContactOptions, [
  { type: undefined, payload: '' },
  { type: 'DERP DERP', payload: '' },
  { type: UPDATE_SETTINGS, payload: badPayload },
  { type: UPDATE_SETTINGS, payload: goodPayload },
])
