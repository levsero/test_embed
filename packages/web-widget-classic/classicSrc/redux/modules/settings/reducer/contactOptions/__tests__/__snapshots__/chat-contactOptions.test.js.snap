// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chatContactOptions, action: DERP DERP 1`] = `
Object {
  "chatLabelOffline": null,
  "chatLabelOnline": null,
}
`;

exports[`chatContactOptions, action: undefined 1`] = `
Object {
  "chatLabelOffline": null,
  "chatLabelOnline": null,
}
`;

exports[`chatContactOptions, action: widget/settings/UPDATE_SETTINGS 1`] = `
Object {
  "chatLabelOffline": null,
  "chatLabelOnline": null,
}
`;

exports[`chatContactOptions, action: widget/settings/UPDATE_SETTINGS 2`] = `
Object {
  "chatLabelOffline": Object {
    "*": "offline label",
  },
  "chatLabelOnline": Object {
    "*": "online label",
  },
}
`;
